internal/configs/transportserver_test.go

```diff
@@ -62,6 +62,164 @@ func TestTransportServerExString(t *testing.T) {
 	}
 }
 
+func TestGenerateTransportServerConfigForTCPSnippets(t *testing.T) {
+	transportServerEx := TransportServerEx{
+		TransportServer: &conf_v1alpha1.TransportServer{
+			ObjectMeta: meta_v1.ObjectMeta{
+				Name:      "tcp-server",
+				Namespace: "default",
+			},
+			Spec: conf_v1alpha1.TransportServerSpec{
+				Listener: conf_v1alpha1.TransportServerListener{
+					Name:     "tcp-listener",
+					Protocol: "TCP",
+				},
+				Upstreams: []conf_v1alpha1.Upstream{
+					{
+						Name:    "tcp-app",
+						Service: "tcp-app-svc",
+						Port:    5001,
+					},
+				},
+				Action: &conf_v1alpha1.Action{
+					Pass: "tcp-app",
+				},
+				Snippets: "deny  192.168.1.1;\nallow 192.168.1.0/24;",
+			},
+		},
+		Endpoints: map[string][]string{
+			"default/tcp-app-svc:5001": {
+				"10.0.0.20:5001",
+			},
+		},
+	}
+
+	listenerPort := 2020
+
+	expected := version2.TransportServerConfig{
+		Upstreams: []version2.StreamUpstream{
+			{
+				Name: "ts_default_tcp-server_tcp-app",
+				Servers: []version2.StreamUpstreamServer{
+					{
+						Address:     "10.0.0.20:5001",
+						MaxFails:    1,
+						FailTimeout: "10s",
+					},
+				},
+				UpstreamLabels: version2.UpstreamLabels{
+					ResourceName:      "tcp-server",
+					ResourceType:      "transportserver",
+					ResourceNamespace: "default",
+					Service:           "tcp-app-svc",
+				},
+			},
+		},
+		Server: version2.StreamServer{
+			Port:                     listenerPort,
+			UDP:                      false,
+			StatusZone:               "tcp-listener",
+			ProxyPass:                "ts_default_tcp-server_tcp-app",
+			Name:                     "tcp-server",
+			Namespace:                "default",
+			ProxyConnectTimeout:      "60s",
+			ProxyNextUpstream:        false,
+			ProxyNextUpstreamTries:   0,
+			ProxyNextUpstreamTimeout: "0s",
+			ProxyTimeout:             "10m",
+			HealthCheck:              nil,
+			Snippets:                 []string{"deny  192.168.1.1;", "allow 192.168.1.0/24;"},
+		},
+	}
+	tsc := &transportServerConfigurator{
+		isPlus:         true,
+		enableSnippets: true,
+	}
+	result := generateTransportServerConfig(&transportServerEx, listenerPort, tsc)
+	if !reflect.DeepEqual(result, expected) {
```

Comment:
Consider using https://pkg.go.dev/github.com/google/go-cmp/cmp#Diff here and in `TestGenerateTransportServerConfigForTCPSnippetsNotEnabled`.

Otherwise fail output can be hard to spot the error, example:
```
--- FAIL: TestGenerateTransportServerConfigForTCPSnippets (0.00s)
    /go/src/github.com/nginxinc/kubernetes-ingress/internal/configs/transportserver_test.go:139: generateTransportServerConfig() returned 
        {Server:{TLSPassthrough:false UnixSocket: Port:2020 UDP:false StatusZone:tcp-listener ProxyRequests:<nil> ProxyResponses:<nil> ProxyPass:ts_default_tcp-server_tcp-app Name:tcp-server Namespace:default ProxyTimeout:10m ProxyConnectTimeout:60s ProxyNextUpstream:false ProxyNextUpstreamTimeout:0s ProxyNextUpstreamTries:0 HealthCheck:<nil> Snippets:[deny  192.168.1.1; allow 192.168.1.0/24;]} Upstreams:[{Name:ts_default_tcp-server_tcp-app Servers:[{Address:10.0.0.20:5001 MaxFails:1 FailTimeout:10s}] UpstreamLabels:{Service:tcp-app-svc ResourceType:transportserver ResourceName:tcp-server ResourceNamespace:default}}]} but expected 
        {Server:{TLSPassthrough:false UnixSocket: Port:2020 UDP:false StatusZone:tcp-listener ProxyRequests:<nil> ProxyResponses:<nil> ProxyPass:ts_default_tcp-server_tcp-app Name:tcp-server Namespace:default ProxyTimeout:10m ProxyConnectTimeout:60s ProxyNextUpstream:false ProxyNextUpstreamTimeout: ProxyNextUpstreamTries:0 HealthCheck:<nil> Snippets:[deny  192.168.1.1; allow 192.168.1.0/24;]} Upstreams:[{Name:ts_default_tcp-server_tcp-app Servers:[{Address:10.0.0.20:5001 MaxFails:1 FailTimeout:10s}] UpstreamLabels:{Service:tcp-app-svc ResourceType:transportserver ResourceName:tcp-server ResourceNamespace:default}}]}
```

Reply:
sure this makes sense